htdocs/product/class/api_products.class.php

```diff
@@ -123,12 +124,19 @@ public function index($sortfield = "t.ref", $sortorder = 'ASC', $limit = 100, $p
         if ($category > 0) {
             $sql.= ", ".MAIN_DB_PREFIX."categorie_product as c";
         }
+        if ($supplier > 0) {
+            $sql.= ", ".MAIN_DB_PREFIX."product_fournisseur_price as s";
```

Comment:
If you add a join with this table, you will return several times the same product if there is several prices for the same supplier.

To keep the philosophy "one table = one CRUD api", i think another API or method should be introduced to return list of product prices of a dedicated supplier.

For example /supplier_prices to get all products + supplier price (so available into. And if you add the filter on third party, you should get the result you want with this new API.

Reply:
@eldy ok! i will update this PR

use this API for marketplace / supplier's products update